sdk/storage/inc/datalake/protocol/datalake_rest_client.hpp

```diff
@@ -0,0 +1,2220 @@
+    }
+    if (!pathCreateOptions.ContentEncoding.empty())
+    {
+      request.AddHeader(k_HeaderContentEncoding, pathCreateOptions.ContentEncoding);
+    }
+    if (!pathCreateOptions.ContentLanguage.empty())
+    {
+      request.AddHeader(k_HeaderContentLanguage, pathCreateOptions.ContentLanguage);
+    }
+    if (!pathCreateOptions.ContentDisposition.empty())
+    {
+      request.AddHeader(k_HeaderContentDisposition, pathCreateOptions.ContentDisposition);
+    }
+    if (!pathCreateOptions.ContentType.empty())
+    {
+      request.AddHeader(k_HeaderContentType, pathCreateOptions.ContentType);
+    }
+    if (!pathCreateOptions.RenameSource.empty())
+    {
+      request.AddHeader(k_HeaderRenameSource, pathCreateOptions.RenameSource);
+    }
+    if (!pathCreateOptions.LeaseIdOptional.empty())
+    {
+      request.AddHeader(k_HeaderLeaseIdOptional, pathCreateOptions.LeaseIdOptional);
+    }
+    if (!pathCreateOptions.SourceLeaseId.empty())
+    {
+      request.AddHeader(k_HeaderSourceLeaseId, pathCreateOptions.SourceLeaseId);
+    }
+    if (!pathCreateOptions.Properties.empty())
+    {
+      request.AddHeader(k_HeaderProperties, pathCreateOptions.Properties);
+    }
+    if (!pathCreateOptions.Permissions.empty())
+    {
+      request.AddHeader(k_HeaderPermissions, pathCreateOptions.Permissions);
+    }
+    if (!pathCreateOptions.Umask.empty())
+    {
+      request.AddHeader(k_HeaderUmask, pathCreateOptions.Umask);
+    }
+    if (!pathCreateOptions.IfMatch.empty())
+    {
+      request.AddHeader(k_HeaderIfMatch, pathCreateOptions.IfMatch);
+    }
+    if (!pathCreateOptions.IfNoneMatch.empty())
+    {
+      request.AddHeader(k_HeaderIfNoneMatch, pathCreateOptions.IfNoneMatch);
+    }
+    if (!pathCreateOptions.IfModifiedSince.empty())
+    {
+      request.AddHeader(k_HeaderIfModifiedSince, pathCreateOptions.IfModifiedSince);
+    }
+    if (!pathCreateOptions.IfUnmodifiedSince.empty())
+    {
+      request.AddHeader(k_HeaderIfUnmodifiedSince, pathCreateOptions.IfUnmodifiedSince);
+    }
+    if (!pathCreateOptions.SourceIfMatch.empty())
+    {
+      request.AddHeader(k_HeaderSourceIfMatch, pathCreateOptions.SourceIfMatch);
+    }
+    if (!pathCreateOptions.SourceIfNoneMatch.empty())
+    {
+      request.AddHeader(k_HeaderSourceIfNoneMatch, pathCreateOptions.SourceIfNoneMatch);
+    }
+    if (!pathCreateOptions.SourceIfModifiedSince.empty())
+    {
+      request.AddHeader(k_HeaderSourceIfModifiedSince, pathCreateOptions.SourceIfModifiedSince);
+    }
+    if (!pathCreateOptions.SourceIfUnmodifiedSince.empty())
+    {
+      request.AddHeader(k_HeaderSourceIfUnmodifiedSince, pathCreateOptions.SourceIfUnmodifiedSince);
+    }
+    return request;
+  }
+
+  struct PathCreateResponse
+  {
+    std::string Date;
+    std::string ETag;
+    std::string LastModified;
+    std::string RequestId;
+    std::string Version;
+    std::string Continuation;
+    uint64_t ContentLength;
```

Comment:
I think compiler will give a warning if you don't give it a default value.

Reply:
ditto for other integer member variables.

Reply:
This is not resolved yet.

Reply:
It is resolved. Please check the latest commit.